main.py

```diff
@@ -256,7 +250,7 @@ def create_plot_box_layout(self, container_id, new_id):
         container = self.manager.get_screen("second").ids[container_id]
         layout = BoxLayout(orientation='vertical')
         container.add_widget(layout)
-        self.manager.get_screen("second").ids[container_id] = layout
+        self.manager.get_screen("second").ids[new_id] = layout
 
```

Comment:
fixed this function. now we dont need the other code. works the same way just cleaner